luxjs/tests/services/pagination.js

```diff
@@ -0,0 +1,185 @@
+    var $lux;
+
+    // create a mock promise that returns data param
+    function createMockPromise(data) {
+        return {
+            get: jasmine.createSpy().and.callFake(function(){
+                return {
+                    then: function(callbacks) {
+                        return callbacks(data);
+                    }
+                }
+            })
+        }
+    }
+
+    beforeEach(function() {
+
+        angular.mock.module('lux.pagination', function($provide) {
+            $provide.service('$lux', function() {
+                this.api = jasmine.createSpy('apiInit');
+                this.api.get = jasmine.createSpy('apiGet');
+            });
+        });
+
+        angular.mock.inject(function(_luxPaginationFactory_, _$lux_) {
+            $lux = _$lux_;
+            LuxPagination = _luxPaginationFactory_;
+        });
+
+    });
+
+    afterEach(function () {
+    });
+
+    it('LuxPagination applies params to itself and set up API', inject(function() {
+        var scope = 'scope';
+        var target = {
+            url: 'originalUrl'
+        }
+        var luxPag = new LuxPagination(scope, target);
+
+        expect(luxPag.scope).toEqual(scope);
+        expect(luxPag.target).toEqual(target);
+        expect(luxPag.orgUrl).toEqual(target.url);
+        expect($lux.api).toHaveBeenCalledWith(target);
+        expect(luxPag.api).toEqual($lux.api(target));
+    }));
+
+    it('LuxPagination applies this.recursive only if parameter passed', function() {
+        var recursive = true;
+        var luxPag = new LuxPagination('target', 'scope', recursive);
+
+        expect(luxPag.recursive).toBe(true);
+        luxPag = new LuxPagination('target','scope');
+        expect(luxPag.recursive).toBe(undefined);
+    });
+
+    it('LuxPagination.getData applies params to itself and calls API', function() {
+        var params = 'params';
+        var cb = jasmine.createSpy('callback');
+        var mockData = {data: true};
+        var luxPag = new LuxPagination('scope', 'target');
+
+        // Fake promise returning good data
+        luxPag.api = createMockPromise(mockData);
+        LuxPagination.prototype.updateUrls = jasmine.createSpy('updateUrls');
+        luxPag.getData(params, cb);
+        expect(luxPag.params).toEqual(params);
+        expect(luxPag.cb).toEqual(cb);
+        expect(luxPag.api.get).toHaveBeenCalledWith(null, params);
+        expect(luxPag.cb).toHaveBeenCalledWith(mockData);
+        expect(luxPag.updateUrls).toHaveBeenCalledWith(mockData);
+    });
+
+    it('LuxPagination.getData passes data into cb and returns if error', function() {
+        var params = 'params';
+        var cb = jasmine.createSpy('callback');
+        var mockError = {error: true};
+        var luxPag = new LuxPagination('scope', 'target');
+
+        // Fake promise returning error data
+        luxPag.api = createMockPromise(mockError);
+        LuxPagination.prototype.updateUrls = jasmine.createSpy('updateUrls');
```

Comment:
It should be possible to use spyOn here rather than jasmine.createSpy though I'm not sure it makes any practical difference in this case.